script-base.js

```diff
@@ -623,6 +772,107 @@ Generator.prototype.dateFormatForLiquibase = function () {
+            break;
+        case 'stripJs' :
+            var regex = '[a-zA-Z]+\:(\s)?\[[ \'a-zA-Z0-9\$\,\(\)\{\}\n\.\<\%\=\>\;\s]*\}\]';
+            //looks for something like mainTranslatePartialLoader: [*]
+            jhipsterUtils.copyWebResource(source, dest, regex, 'js', _this, _opt, template);
+            break;
+        case 'copy' :
+            _this.copy(source, dest);
+            break;
+        default:
+            _this.template(source, dest, _this, _opt);
+    }
+}
+
+/**
+ * Copy html templates after stripping translation keys when translation is disabled.
+ *
+ * @param {source} path of the source file to copy from
+ * @param {dest} path of the destination file to copy to
+ * @param {_this} context that can be used as the generator instance or data to process template
+ * @param {_opt} options that can be passed to template method
+ * @param {template} flag to use template method instead of copy
+ */
+Generator.prototype.copyHtml = function (source, dest, _this, _opt, template) {
+    this.copyTemplate(source, dest, 'stripHtml', _this, _opt, template);
+}
+
+/**
+ * Copy Js templates after stripping translation keys when translation is disabled.
+ *
+ * @param {source} path of the source file to copy from
+ * @param {dest} path of the destination file to copy to
+ * @param {_this} context that can be used as the generator instance or data to process template
+ * @param {_opt} options that can be passed to template method
+ * @param {template} flag to use template method instead of copy
+ */
+Generator.prototype.copyJs = function (source, dest, _this, _opt, template) {
+    this.copyTemplate(source, dest, 'stripJs', _this, _opt, template);
+}
+
+/**
+ * Rewrite the specified file with provided content at the needle location
+ *
+ * @param {fullPath} path of the source file to rewrite
+ * @param {needle} needle to look for where content will be inserted
+ * @param {content} content to be written
+ */
+Generator.prototype.rewriteFile = function(filePath, needle, content) {
+    try {
+        jhipsterUtils.rewriteFile({
+            file: filePath,
+            needle: needle,
+            splicable: [
+              content
+            ]
+        });
+    } catch (e) {
+        console.log(chalk.yellow('\nUnable to find ') + filePath + chalk.yellow(' or missing required needle. File rewrite failed.\n'));
+    }
+};
+
+/**
+ * Replace the pattern/regex with provided content
+ *
+ * @param {fullPath} path of the source file to rewrite
+ * @param {pattern} pattern to look for where content will be replaced
+ * @param {content} content to be written
+ * @param {regex} true if pattern is regex
+ */
+Generator.prototype.replaceContent = function(filePath, pattern, content, regex) {
+    try {
+        jhipsterUtils.replaceContent({
+            file: filePath,
+            pattern: pattern,
+            content: content,
+            regex, regex
```

Comment:
I don't test and my knowledge in js is quite low but it's normal? it's not to be `regex: regex`?